From 042958ff8fe4ff5907a523f1f94de4f7eb485232 Mon Sep 17 00:00:00 2001
From: Dmitriy Chumak <dmitriy_chumak@xyratex.com>
Date: Tue, 2 Apr 2013 11:18:02 +0300
Subject: [PATCH] scripts/m0: add 'rpm' command

Introduce new 'rpm' command in m0 script to create Mero rpm package.
Also, include this command into 'check-everything' command.
---
 scripts/m0 | 31 +++++++++++++++++++++++++++----
 1 file changed, 27 insertions(+), 4 deletions(-)

diff --git a/scripts/m0 b/scripts/m0
index c17c880..4e4c8fc 100755
--- a/scripts/m0
+++ b/scripts/m0
@@ -19,7 +19,9 @@ SRC="${SRC%/*/*}"
 
 SUDO='sudo -E'
 
-die() { echo "$@" >&2; exit 1; }
+echo_err() { echo "$@" >&2; }
+
+die() { echo_err "$@"; exit 1; }
 
 clean() {
     cd "$SRC"
@@ -94,11 +96,28 @@ dist_check() {
     cd -
     rm -r $DIST
 
-    [ "$1" = '-a' ] || rm $DISTFILE
+    [ "$1" = '-a' ] && rm $DISTFILE
+}
+
+rpm_check() {
+    local DISTFILE='mero-0.1.0.tar.gz'
+
+    cd "$SRC"
+    [ ! -f $DISTFILE ] && {
+        echo_err "WARNING: dist archive $DISTFILE not found, please create" \
+                 'it with `make dist` command first'
+        return
+    }
+
+    rpmbuild --define "_sourcedir $(pwd)" -bb mero.spec
+
+    local RPMFILE=$(ls $HOME/rpmbuild/RPMS/x86_64/mero*)
+
+    echo "Resulting rpm file: $RPMFILE"
 }
 
 check_everything() {
-    for cmd in clean build dist_check run_ut run_kut run_st; do
+    for cmd in clean build dist_check rpm_check run_ut run_kut run_st; do
 	echo "----- $cmd -----" >&2
 	$cmd
     done
@@ -126,9 +145,12 @@ Commands:
     run-all             Run unit and system tests.
 
     dist-check [-a]     Create and validate Mero distribution archive.
-                        The resulting archive file will be deleted,
+                        The resulting archive file will not be deleted,
                         unless '-a' option is used.
 
+    rpm                 Create Mero rpm package from existing dist
+                        archive.
+
     check-everything  = (rebuild + dist-check + run-all) * 2
                         Please run this command before landing.
                         .
@@ -164,6 +186,7 @@ case "$1" in
         run_kut
         run_st;;
     dist-check) shift; dist_check "$@";;
+    rpm) rpm_check;;
     check-everything)
 	check_everything
 	echo $CONFIGURE_OPTS | grep -qe '--disable-m0-asserts' || {
-- 
1.8.3.2

